### DISCUSSION TITLE
Budgets and Start and End Dates

### DISCUSSION THREAD

**Discussion by soloam** [firefly-iii/firefly-iii#8756] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/explanation/more-information/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

I know that a month to firefly is from 1 to the last day of the month... But I'm able to adjust to see what I need with the date picker. The only problem is the budgets! Is it possible to select the dates to  begin and end for the budgets? The current fixed days makes it useless to my use case, my income is ate the 21st, and I would like to track budgets from that day. At the 1st I have a lot of expenses that are not taken into consideration on the budget.

### Solution

Allow to select the first and last day of the budgets... Even If that means that They can't be automatically created!

### What are alternatives?

Anything that allows to change the dates of the budgets!

### Additional context

No

**Comment by JC5** [comment]
Hey, thanks for the suggestion. Although you can create budgets with any date range using the API, the user interface will not get this as an option.

https://docs.firefly-iii.org/explanation/firefly-iii/background/personal-finances/

Everybody gets paid halfway during the month. It's pretty rare to be paid on the first. So my suggestion is to turn your finances around so you get a monthly rhythm, regardless of when you get paid. 

That's easier to budget, easier to manage and fits right into Firefly III.
